### ISSUE TITLE
Cannot build ViewableInventory

### ISSUE THREAD

**Issue by mosemister** [SpongePowered/Sponge#3545]
**I am currently running**

- SpongeVanilla version:  https://github.com/SpongePowered/Sponge/actions/runs/1281098566
- Java version: 1.16
- Operating System: Win10

- Plugins/Mods:
- Custom mod
- 

**Issue Description**

```text
java.lang.IllegalStateException: Plugin has not been set in this builder!
```

The above error occurs before I can provide a plugin to the builder. 

This is probably due to this https://github.com/SpongePowered/Sponge/blob/ece3eb6b59556e1159ae3baca15bef703e5fea73/src/main/java/org/spongepowered/common/inventory/custom/SpongeViewableInventoryBuilder.java#L240 missing a plugin provider

```text
        at org.spongepowered.common.inventory.SpongeInventoryBuilder.build(SpongeInventoryBuilder.java:118)
        at org.spongepowered.common.inventory.custom.SpongeViewableInventoryBuilder.completeStructure(SpongeViewableInventoryBuilder.java:240)
```

My code
```java
ViewableInventory.builder().type(type).completeStructure()
```

**Comment by hYdos** [comment]
It's telling you. `Plugin has not been set in this builder`.
Is there a method called plugin or a method which takes a plugin in the builder?

**Comment by mosemister** [comment]
Im well aware of what it says. The ``plugin`` method is after the point in my code (my code is returning a ``EndStep`` with you supplying the pluginContainer to the EndStep). 

The issue is I cant create the endstep unless a plugin is supplied. 

After viewing the error in more detail I found that Sponge internally creates a none viewable inventory, however doesnt provide a plugin to the internal endstep (which I had provided a link directly to originally) which seems to be the cause of the crash
